app/controllers/api/v1/accounts/statuses_controller.rb

```diff
@@ -58,6 +59,10 @@ def no_replies_scope
     Status.without_replies
   end
 
+  def filter_with_visibility_scope(visibility)
+    Status.where(visibility: Status.visibilities[visibility])
```

Comment:
You can't do this because it would likely conflict with Status#permitted_for and possibly leak statuses you aren't authorized to view.